File: shared/src/main/java/ch/zxseitz/tbsg/games/IGame.java
package ch.zxseitz.tbsg.games;

import java.util.Collection;

public interface IGame<T> extends Comparable<IGame<?>> {
    String getId();
    int getNext();
    GameState getState();
    int[] getBoard();
    Collection<T> getPreview();
    void update(T action);
}

File: shared/src/main/java/ch/zxseitz/tbsg/games/IGame.kt
package ch.zxseitz.tbsg.games;

import ch.zxseitz.tbsg.games.exceptions.GameException;
import kotlin.jvm.Throws

interface IGame<T>: Comparable<IGame<*>> {
    val id: String
    val next: Int
    val state: GameState
    val board: IntArray
    val preview: Collection<T>
    fun init();
    @Throws(GameException::class)
    fun update(action: T)
}
